# Combining Data
After collecting about a million comments, it's time to compile them. We'll remove duplicates based on Comment ID.

In [2]:
import pandas as pd
from utils import parse_yt_datetime, parse_and_concat_csvs

In [3]:
# Depending on the features you were interested in, your headers may or may not look the same
header = ['Comment', 'Comment ID', 'Replies Count', 'Like Count', 'Updated At']

In [4]:
csvs = ['aya_blackpink_time_8.csv', 'aya_blackpink_time_9.csv', 'aya_blackpink_time_10.csv', 
        'aya_blackpink_time_11.csv', 'aya_blackpink_time_12.csv', 'aya_blackpink_time_13.csv', 
        'aya_blackpink_time_14.csv']


In [ ]:
combined = parse_and_concat_csvs(csvs, header)

In [85]:
combined.shape

(1143597, 5)

# Sort by Comment ID
Not by the text content of the comments, since commenters may write the same things.

In [123]:
combined.sort_values("Comment ID", inplace = True) 
  


In [124]:
len(combined['Comment ID'])-len(combined['Comment ID'].drop_duplicates())

652844

652844 is the number of duplicate comments we have in our csv files.

In [125]:
combined.duplicated(subset='Comment ID', keep='first').sum()

652844

In [126]:
removed_duplicates = combined.drop_duplicates(subset="Comment ID", keep='first', inplace=False)

In [127]:
sort_by_update_time = removed_duplicates.sort_values("Updated At")

In [128]:
sort_by_update_time.reset_index()

index                                            Comment  \
0       40295                                              😂😂😂😂😂   
1       40297  ستريم ستريم ستريم نبي نوصل ٨٢ قبل مايخلص اليوم...   
2       40298                81M congratulations blink blackpink   
3       40299                                       #1 in Poland   
4       40296                                    QUEEN BLACKPINK   
5       40291                                   After 1 year:/🤲🏻   
6       40293                                        New record.   
7       40294  Bp broke the record for most viewed mv in 24/h...   
8       40292            ดูอีกกี่รอบ ก็ใช้คำว่าสวยได้เปลืองมาก 😄   
9       40288                       100m view in 24hour please😍😍   
10      40290  Keep streaming Blink. 85m in 24h is waiting fo...   
11      40285  Queens ❤️ wow 81 million views in 23 hours ❤️❤...   
12      40286                                                81M   
13      40284                                                  5   
14      40283                                      Я в безумее 😢   
15      40282  We have 40mins c'mon BLINKS we can get it to 8...   
16      40287                       https://youtu.be/rmZ9dTwkZ_8   
17      40276          Mizo eng zat takin views ve ang mo???????   
18      40277                                     Congratzzzzzzz   
19      40280                   Let’s get 6 million likes blinks   
20      40278                                          Tuyệt vời   
21      40279                                             85M 👆👆   
22      40281  Jiso : so beautiful and georgeous\nJenny: hot\...   
23      40275        DUM DUM DUM DUM DUM  DUM  HOW YOU LIKE THAT   
24      40274                                   I love Blackpink   
25      40267               💕💕💕💕💕💕💕💕💕💕💕💕💕💕💕💕💕\n💕💕💕💕💕💕💕💕💕💕💕💕💕💕💕💕💕   
26      40266  Lets make this 100m views w/in 24 hours we can...   
27      40268                                                81M   
28      40272  11 million views in 4 hours\nWe have proved th...   
29      40265                                   Blackpink's Era.   
...       ...                                                ...   
490723     28  Aku suka banget blackpink tau suka banget dari...   
490724     27  We can't break the record for the fastest to r...   
490725     26  https://www.youtube.com/channel/UCJmSZExD7ItHK...   
490726     25                                     Lisa so pretty   
490727     24               Alguien que hable español por aqui👐🤣   
490728     23                   Irene : but , im original visual   
490729     22                                           off cors   
490730     21                                              يموجي   
490731     20                                              يموجي   
490732     19                                 HYLT 2ND WIN!! YAY   
490733     18                                              يموجي   
490734     17                                              يموجي   
490735     16                                       €£¥₩♡#☆¿€¥£¥   
490736     15                                              يموجي   
490737     14  لحد الان انا لم اجد اي تعليق عربي سو عجبتني كت...   
490738     13                                              يموجي   
490739     12  woow 😍😍😍😍♥️♥️♥️♥️♥️🌞🌞🌞🌞🌞🌞🎀🎀🎀🎀🎀🎀🎈🎈🎈🎈🎈😃😃😃😃😃😃😃😋😋😋...   
490740     11                                              يموجي   
490741     10  Guys pls watch dis 1 its a jolly reaction and ...   
490742      9                              Salve irmões do 8 ano   
490743      8                                 blackpink let's go   
490744      7           Omg 41M subs <3 Congarts  !! LOVE YOU <3   
490745      6  🖕🖕🖕🖕🖕🖕🖕🖕آشغالاي خر\nدذد هاي کثيف 🖕🖕🖕🖕🖕🖕🖕🖕🖕🖕🖕🖕🖕...   
490746      5             ❤️💛💜🖤💚💙💗❤️💖💕💝😆😘♥️🌞🎀🎈😋😋🌚🌚🌚🙂😗😩😝🙃😁🌞🎀🎈♥️♥️   
490747      4                           Love u so much JENNIE AH   
490748    568  Haters: talking trash about blackpink \nWhile ...   
490749      3       Tui bị bay acc tiktok vs lai dt mat r 😄😊💔💔💔💔

In [129]:
sort_by_update_time.shape

(490753, 5)

In [130]:
sort_by_update_time.to_csv('comments_by_time_june27_july08.csv', index=False)

In [132]:
testing = pd.read_csv('comments_by_time_june27_july08.csv', lineterminator='\n')

In [136]:
testing.shape

(490753, 5)

In [135]:
testing.head()

Comment  \
0                                              😂😂😂😂😂   
1  ستريم ستريم ستريم نبي نوصل ٨٢ قبل مايخلص اليوم...   
2                81M congratulations blink blackpink   
3                                       #1 in Poland   
4                                    QUEEN BLACKPINK   

                   Comment ID  Replies Count  Like Count            Updated At  
0  UgyJUozOOb_ng7rmIMp4AaABAg              0           0  2020-06-27T08:20:15Z  
1  Ugw8PM1u0bsWOMzzGCZ4AaABAg              1           0  2020-06-27T08:20:15Z  
2  Ugymm_tQHxSp7U2wH8t4AaABAg              0           1  2020-06-27T08:20:15Z  
3  UgwymZEm7y2vyppPyCh4AaABAg              0           1  2020-06-27T08:20:15Z  
4  UgzotV8vjKKQonnXCUN4AaABAg              0           1  2020-06-27T08:20:15Z